test/basicfuns.jl

```diff
@@ -9,14 +9,19 @@ using StatsFuns, Test
     @test iszero(xlogy(0, 1))
     @test xlogy(2, 3) ≈ 2.0 * log(3.0)
     @test_throws DomainError xlogy(1, -1)
-    # we allow negative `x`, https://github.com/JuliaStats/StatsFuns.jl/pull/95#discussion_r427558736
-    @test xlogy(-2, 3) == -xlogy(2, 3)
     @test isnan(xlogy(NaN, 2))
     @test isnan(xlogy(2, NaN))
+
+    # Since we allow complex/negative values, test for them. See comments in:
```

Comment:
Sorry to be a pain, but I'm also concerned with corner cases like those involving a complex and a zero, as well as `Complex(Inf, NaN)` and so on.

Reply:
For all those cases it should just return whatever `x * log(y)` turns out to give (which is the correct answer). The only special case is `x = 0` and y != NaN`. Anyway I added all the tests tests I could think of (if you have any special case in mind I can add it as well).